### PR TITLE
Catroid-94 Incorrect handling of deletion of variables and lists

### PR DESCRIPTION
**Pull request by BeinhAlex**
https://jira.catrob.at/browse/CATROID-94

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/brickspinner/BrickSpinner.java [hunk 1]
```diff
@@ -142,7 +142,7 @@ public Object getSelection() {
 	}
 
 	private int consolidateSpinnerSelection(int position) {
-		if (position == -1) {
+		/*if (position == -1) {
```
Comment (DinosaurierRex) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/brickspinner/BrickSpinner.java:145]: Please remove comments

**Review by DinosaurierRex — CHANGES_REQUESTED** [review]
Please have a look at the comments.

**Review by DinosaurierRex — APPROVED** [review]
LGTM now 👍

**Review by wslany — CHANGES_REQUESTED** [review]
The issue was fixed for global variables and lists, but it still exists for local variables and local lists (available only in the current object) as well as also still for multiplayer variables. Should these be handled in a different ticket? Note that it is more complex for local variables and local lists, since the scope of conflicts is limited to the current object.

Also, it seems that there are some issues with some of the unit tests. As far as I see, they are not being executed on Jenkins, is that correct? If they run through correctly locally, can you please add screenshots? Otherwise, please investigate what the issue could be.

**Comment by wslany** [comment]
One more case where the "still in use" warning should be shown:
- in the formula editor, tap on "Data" and create a new variable
- tap on the new variable to add it to the formula 
- while still editing the formula, tap again on Data and then delete the variable 
- observe that no warning is shown, the variable is deleted
- when you return to the formula from the Data screen, the variable is still included in the formula.

**Pull request closed without merging**